connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SQLImplicitsTestSuite.scala

```diff
@@ -130,9 +131,15 @@ class SQLImplicitsTestSuite extends ConnectFunSuite with BeforeAndAfterAll {
     testImplicit(BigDecimal(decimal))
     testImplicit(Date.valueOf(LocalDate.now()))
     testImplicit(LocalDate.now())
-    testImplicit(LocalDateTime.now())
-    testImplicit(Instant.now())
-    testImplicit(Timestamp.from(Instant.now()))
+    // SPARK-42770: Run `LocalDateTime.now()` and `Instant.now()` with Java 8 & 11 always
+    // get microseconds on both Linux and MacOS, but there are some differences when
+    // using Java 17, it will get accurate nanoseconds on Linux, but still get the microseconds
+    // on MacOS. At present, Spark always converts them to microseconds, this will cause the
+    // test fail when using Java 17 on Linux, so add `truncatedTo(ChronoUnit.MICROS)` to
+    // there to ensure the accuracy of input data is microseconds.
+    testImplicit(LocalDateTime.now().truncatedTo(ChronoUnit.MICROS))
+    testImplicit(Instant.now().truncatedTo(ChronoUnit.MICROS))
+    testImplicit(Timestamp.from(Instant.now().truncatedTo(ChronoUnit.MICROS)))
```

Comment:
Shall we apply this change for Java 17 only?

Reply:
Also, it would be great if we can apply new code on Linux environment only, @LuciferYang .